indexing-service/src/main/java/org/apache/druid/indexing/common/config/TaskConfig.java

```diff
@@ -182,6 +197,16 @@ public TaskConfig(
     }
     log.debug("Batch processing mode:[%s]", this.batchProcessingMode);
     this.storeEmptyColumns = storeEmptyColumns == null ? DEFAULT_STORE_EMPTY_COLUMNS : storeEmptyColumns;
+
+    this.baseTaskDir = baseTaskDir;
+    if (CollectionUtils.isNullOrEmpty(baseTaskDirPaths)) {
```

Comment:
I think overriding baseTaskDir(Path) when baseTaskDirPaths is provided might be better as it is done in other places of the code like javaOpts and javaOptsArray.

Reply:
Can you please respond to this for closure?

Reply:
Certain deployments have default configurations which are already set. If baseTaskDir is one of them, it would benefit them if the property could be overridden with baseTaskDirPaths without throwing an exception